public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -265,6 +265,10 @@ export class LokiDatasource extends DataSourceApi<LokiQuery, LokiOptions> {
         selector = addLabelToSelector(selector, action.key, action.value);
         break;
       }
+      case 'ADD_FILTER_OUT': {
```

Comment:
What does this do if we already have that label `{label="value"}` will it add `{label="value", label!="value"}`?

Reply:
Yes, that is exactly what it currently does. I have checked elastic's kibana and they have the same functionality -> so it results in no logs, because there are no logs that have and doesn't have specific value for label at the same time.

But I think it would make sense to add there a logic -> remove the contradicting = or != conditionality.

Reply:
I think that's fine for now. Should really be up for the datasource to figure out what to do with conflicting user input.